docs/tutorials/built-your-own.md

```diff
@@ -0,0 +1,175 @@
+)
+
+func main() {
+	cfg := config.NewConfig()
+	if err := cfg.Parse(); err != nil {
+		log.Fatalf("Error processing config: %s", err)
+	}
+
+	log.SetLevel(cfg.ApplicationLogLevel)
+
+	opt := cfg.ToOptions()
+	opt.CustomFilters = append(opt.CustomFilters, lfilters.NewMyFilter())
+
+	log.Fatal(skipper.Run(opt))
+}
+```
+
+## Code
+Write the code and use the custom filter implemented in https://github.com/szuecs/skipper-example-proxy/blob/main/filters/custom.go
+```
+[:~]% mkdir -p /tmp/go/skipper
+[:~]% cd /tmp/go/skipper
+[:/tmp/go/skipper]% go mod init myproject
+go: creating new go.mod: module myproject
+[:/tmp/go/skipper]% cat >main.go
+package main
+
+import (
+	log "github.com/sirupsen/logrus"
+	lfilters "github.com/szuecs/skipper-example-proxy/filters"
+	"github.com/zalando/skipper"
+	"github.com/zalando/skipper/config"
+)
+
+func main() {
+	cfg := config.NewConfig()
+	if err := cfg.Parse(); err != nil {
+		log.Fatalf("Error processing config: %s", err)
+	}
+
+	log.SetLevel(cfg.ApplicationLogLevel)
+
+	opt := cfg.ToOptions()
+	opt.CustomFilters = append(opt.CustomFilters, lfilters.NewMyFilter())
+
+	log.Fatal(skipper.Run(opt))
+}
+CTRL-D
+[:/tmp/go/skipper]%
+```
+
+## Build
+I had some import failures because of a transitive dependency.
+```
+[:/tmp/go/skipper]% go mod tidy
+go: finding module for package github.com/zalando/skipper/config
+go: finding module for package github.com/szuecs/skipper-example-proxy/filters
+go: finding module for package github.com/sirupsen/logrus
+go: finding module for package github.com/zalando/skipper
+go: found github.com/sirupsen/logrus in github.com/sirupsen/logrus v1.9.3
+go: found github.com/szuecs/skipper-example-proxy/filters in github.com/szuecs/skipper-example-proxy v0.0.0-20230622190245-63163cbaabc8
+go: found github.com/zalando/skipper in github.com/zalando/skipper v0.16.117
+go: found github.com/zalando/skipper/config in github.com/zalando/skipper v0.16.117
+go: finding module for package github.com/nxadm/tail
+go: finding module for package github.com/kr/text
+go: finding module for package github.com/rogpeppe/go-internal/fmtsort
+go: found github.com/kr/text in github.com/kr/text v0.2.0
+go: found github.com/rogpeppe/go-internal/fmtsort in github.com/rogpeppe/go-internal v1.10.0
+go: found github.com/nxadm/tail in github.com/nxadm/tail v1.4.8
+myproject imports
+        github.com/zalando/skipper imports
+        github.com/zalando/skipper/filters/auth tested by
+        github.com/zalando/skipper/filters/auth.test imports
+        golang.org/x/oauth2/google imports
+        cloud.google.com/go/compute/metadata: ambiguous import: found package cloud.google.com/go/compute/metadata in multiple modules:
+        cloud.google.com/go v0.65.0 (/home/sszuecs/go/pkg/mod/cloud.google.com/go@v0.65.0/compute/metadata)
+        cloud.google.com/go/compute/metadata v0.2.0 (/home/sszuecs/go/pkg/mod/cloud.google.com/go/compute/metadata@v0.2.0)
+zsh: exit 1     go mod tidy
```

Comment:
@AlexanderYastrebov This will be fixed in https://go.dev/cl/496635 so go >=1.20.6